salt/transport/tcp.py

```diff
@@ -656,114 +373,28 @@ def post_fork(self, payload_handler, io_loop):
                         (self.opts["interface"], int(self.opts["ret_port"]))
                     )
                 self.req_server = SaltMessageServer(
-                    self.handle_message,
+                    message_handler,
                     ssl_options=self.opts.get("ssl"),
-                    io_loop=self.io_loop,
+                    io_loop=io_loop,
                 )
                 self.req_server.add_socket(self._socket)
                 self._socket.listen(self.backlog)
-        salt.transport.mixins.auth.AESReqServerMixin.post_fork(
-            self, payload_handler, io_loop
-        )
 
-    @salt.ext.tornado.gen.coroutine
-    def handle_message(self, stream, header, payload):
-        """
-        Handle incoming messages from underlying tcp streams
-        """
-        try:
-            try:
-                payload = self._decode_payload(payload)
-            except Exception:  # pylint: disable=broad-except
-                stream.write(salt.transport.frame.frame_msg("bad load", header=header))
-                raise salt.ext.tornado.gen.Return()
-
-            # TODO helper functions to normalize payload?
-            if not isinstance(payload, dict) or not isinstance(
-                payload.get("load"), dict
-            ):
-                yield stream.write(
-                    salt.transport.frame.frame_msg(
-                        "payload and load must be a dict", header=header
-                    )
-                )
-                raise salt.ext.tornado.gen.Return()
+    def wrap_stream(self, stream):
```

Comment:
Newbie question: does this code need to run on python 2.X? If not, should we start moving toward strongly-typed arguments?

Reply:
Python doesn't have strong typing. We do have [type hints](https://docs.python.org/3/library/typing.html#module-typing), but those are used by third party per-processors at compile time (not runtime). We are not currently running anything that checks type hits at the moment. That would be a valuable addition though.

Reply:
Right. I meant to say "type hints". Thanks.